akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -346,9 +349,12 @@ final class Split[T](decision: Split.SplitDecision, p: T ⇒ Boolean) extends Gr
             closeThis(handler, elem)
             handOver(handler)
           } else {
-            // Drain into the void
-            if (substreamCancelled) pull(in)
-            else substreamSource.push(elem)
+            if (substreamCancelled) {
+              if (eagerClose) completeStage()
```

Comment:
> I've seen both being used in different stages/combinators, is there a clear distinction between them? eagerCancel is for behaviour related to downstream completion and eagerClose is for upstream, correct?

We have renamed that parameter on the Merge operations to be more precise: there is it about emitting the completion signal, here it is about emitting the cancellation signal.